script/tool/lib/src/gradle_check_command.dart

```diff
@@ -228,24 +241,31 @@ apply plugin: "com.google.cloud.artifactregistry.gradle-plugin"
         r'classpath.*gradle\.plugin\.com\.google\.cloud\.artifactregistry:artifactregistry-gradle-plugin');
     final RegExp artifactRegistryPluginApplyRegex = RegExp(
         r'apply.*plugin.*com\.google\.cloud\.artifactregistry\.gradle-plugin');
+    final RegExp artifactRegistryPluginApplyRegexGP = RegExp(
+        r'id.*com\.google\.cloud\.artifactregistry\.gradle-plugin.*version.*\b\d+\.\d+\.\d+\b');
+
 
     final bool documentationPresent = gradleLines
         .any((String line) => documentationPresentRegex.hasMatch(line));
     final bool artifactRegistryDefined = gradleLines
         .any((String line) => artifactRegistryDefinitionRegex.hasMatch(line));
     final bool artifactRegistryPluginApplied = gradleLines
         .any((String line) => artifactRegistryPluginApplyRegex.hasMatch(line));
+    final bool artifactRegistryPluginAppliedGP = gradleLines
+        .any((String line) => artifactRegistryPluginApplyRegexGP.hasMatch(line));
+
+    final bool validArtifactConfiguration = documentationPresent &&
+        ((artifactRegistryDefined && artifactRegistryPluginApplied) || artifactRegistryPluginAppliedGP);
 
-    if (!(documentationPresent &&
-        artifactRegistryDefined &&
-        artifactRegistryPluginApplied)) {
+    if (!validArtifactConfiguration && !artifactRegistryPluginAppliedGP) {
+      printError('Failed Artifact Hub validation. Include the following in '
+          'example root settings.gradle:\n$exampleSettingsArtifactHubStringGP');
+    }
```

Comment:
Instead of these lines which I think are a response to [this thread](https://github.com/flutter/packages/pull/7822/files#r1819449426). 

add documentationPresent to the check for GP method of applying artifact registry.

Reply:
done